R/RtGam.R

```diff
@@ -0,0 +1,28 @@
+#' Stub: Will become fitting/class constructor
+#'
+#' Including as a stub function to showcase control flow
+#' @param cases A vector of non-negative incident case counts occurring on an
+#'   associated `reference_date`. Missing values (NAs) are not allowed.
+#' @param reference_date The associated date on which the count of incident
+#'   `cases` occurred. Missing dates are not allowed and dates can only occur
+#'   once.
+#' @param group The geographic grouping for the case/reference-date pair. Not
+#' yet implemented and a value other than `NULL` will throw an error.
+#'
+#' @returns Stub function: NULL
+#' @export
+#' @examples
+#' cases <- c(1, 2, 3)
+#' reference_date <- as.Date(c("2023-01-01", "2023-01-02", "2023-01-03"))
+#' mod <- RtGam::RtGam(cases, reference_date)
+RtGam <- function(cases,
```

Comment:
Do you see a clear value in making the input be vectors vs a dataframe of cases w/ dates?

Reply:
> the decision to use an input that has a vector for cases and dates vs a dataframe

It's an excellent question and one I went back and forth on. I think there's a case to be made for either approach. Using a dataframe enforces for the user that the relationship between case/date/group is 1:1 and we have to construct a dataframe for mgcv anyway.

But I also find the dataframe workflow a bit clunky. It requires enforcing particular column names and these are invariably a bit different tool-to-tool. And internally, we need to break apart the dataframe into vectors anyway to do checking on each component. We can short-circuit that process by requesting vectors in the first place, which I think leads to a slightly cleaner pattern.

> failing that no concept of dates (so you can use any time step without internal modification)

This is a another good q. The motivation here is twofold:

1. I always find the date -> timestep conversion a bit tricky to get just right. We want to use small numbers (and probably z-score?) and I haven't found a super easy way to do that in R. That's the kind of thing a package with tested code is helpful with.
2.  We need to preserve that specific conversion for later prediction on arbitrary dates, so I wanted to handle it internally and get it right once.

Reply:
> It requires enforcing particular column names and these are invariably a bit different tool-to-tool. 

For this approach, I have been using a class constructor to prepare data. Agree it is still a little clunky but also pretty standard. 

> And internally, we need to break apart the dataframe into vectors anyway to do checking on each component. 

I'm not sure this is true if using tools for `checkmate` etc.